File: android-native/app/src/main/java/com/example/kpkn/screens/wikilab/TendonDetailScreen.kt
package com.example.kpkn.screens.wikilab

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.input.nestedscroll.nestedScroll
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.kpkn.data.db.TendonEntity
import com.example.kpkn.data.repository.InjuryInfo
import com.example.kpkn.data.repository.WikiLabRepository

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun TendonDetailScreen(
    tendonId: String,
    onNavigateToMuscle: (String) -> Unit,
    onNavigateToJoint: (String) -> Unit,
    onNavigateToExercise: (String) -> Unit,
    onBack: () -> Unit,
) {
    val tendon = WikiLabRepository.getTendonById(tendonId)

    if (tendon == null) {
        Box(Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            Text("Tendón no encontrado")
        }
        return
    }

    val injuries = WikiLabRepository.parseInjuries(tendon.commonInjuries)
    val protectionIds = WikiLabRepository.parseStringList(tendon.protectiveExercises)
    val protectiveExercises = remember(protectionIds) {
        resolveWikiLabExerciseLinks(protectionIds)
    }
    val tendonGuide = remember(tendon.id) { buildTendonGuide(tendon) }
    val scrollBehavior = TopAppBarDefaults.exitUntilCollapsedScrollBehavior()

    Scaffold(
        containerColor = Color.Black,
        modifier = Modifier.nestedScroll(scrollBehavior.nestedScrollConnection),
        topBar = {
            LargeTopAppBar(
                title = {
                    Column {
                        Text(tendon.name, fontWeight = FontWeight.Black, fontFamily = FontFamily.Serif, color = Color.White)
                        Surface(
                            shape = RoundedCornerShape(8.dp),
                            color = Color(0xFFFF8F00).copy(alpha = 0.12f),
                        ) {
                            Text(
                                "Tendon",
                                modifier = Modifier.padding(horizontal = 10.dp, vertical = 4.dp),
                                style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif),
                                fontWeight = FontWeight.Bold,
                                color = Color(0xFFFF8F00),
                            )
                        }
                    }
                },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.Default.ArrowBack, "Volver", tint = Color.White)
                    }
                },
                colors = TopAppBarDefaults.largeTopAppBarColors(
                    containerColor = Color.Black,
                    scrolledContainerColor = Color.Black,
                    titleContentColor = Color.White,
                    navigationIconContentColor = Color.White,
                ),
                scrollBehavior = scrollBehavior,
            )
        },
    ) { padding ->
        LazyColumn(
            modifier = Modifier.fillMaxSize().background(Color.Black).padding(padding),
            contentPadding = PaddingValues(horizontal = 16.dp, vertical = 8.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp),
        ) {
            // ─── Header ──────────────────────────────────────────────────
            item {
                Column {
                    tendon.description?.let {
                        Spacer(Modifier.height(2.dp))
                        Text(
                            it,
                            style = MaterialTheme.typography.bodyMedium.copy(
                                fontFamily = FontFamily.Serif,
                                color = Color.White.copy(alpha = 0.9f),
                            ),
                            lineHeight = 22.sp,
                        )
                    }
                }
            }

            item {
                WikiLabInsightCard(
                    title = tendonGuide.title,
                    accent = tendonGuide.accent,
                    icon = tendonGuide.icon,
                    summary = tendonGuide.summary,
                    bullets = tendonGuide.bullets,
                )
            }

            // ─── Related Muscle ──────────────────────────────────────────
            tendon.muscleId?.let { muscleId ->
                val canonicalMuscleId = canonicalWikiLabMuscleIdFromEntityId(muscleId) ?: muscleId
                val muscle = WikiLabRepository.getMuscleById(canonicalMuscleId)
                if (muscle != null) {
                    item {
                        Column(
                            modifier = Modifier
                                .fillMaxWidth()
                                .clip(RoundedCornerShape(12.dp))
                                .background(Color(0xFF121212))
                                .border(BorderStroke(1.dp, Color(0xFF1E1E1E)), RoundedCornerShape(12.dp))
                                .clickable { onNavigateToMuscle(canonicalMuscleId) }
                                .padding(16.dp),
                        ) {
                            Row(verticalAlignment = Alignment.CenterVertically) {
                                Icon(Icons.Default.FitnessCenter, null, Modifier.size(14.dp), tint = Color(0xFF9C27B0))
                                Spacer(Modifier.width(6.dp))
                                Text(
                                    "MÚSCULO RELACIONADO",
                                    style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif),
                                    fontWeight = FontWeight.ExtraBold,
                                    letterSpacing = (0.1f).sp,
                                    color = Color.White.copy(alpha = 0.5f),
                                )
                            }
                            Spacer(Modifier.height(8.dp))
                            Text(muscle.name, style = MaterialTheme.typography.titleMedium.copy(fontFamily = FontFamily.Serif), fontWeight = FontWeight.Bold, color = Color(0xFF9C27B0))
                            Text(muscle.description.take(100) + "...", style = MaterialTheme.typography.bodySmall.copy(fontFamily = FontFamily.Serif), color = Color.White.copy(alpha = 0.7f))
                        }
                    }
                }
            }

            // ─── Related Joint ───────────────────────────────────────────
            tendon.jointId?.let { jointId ->
                val joint = WikiLabRepository.getJointById(jointId)
                if (joint != null) {
                    item {
                        Column(
                            modifier = Modifier
                                .fillMaxWidth()
                                .clip(RoundedCornerShape(12.dp))
                                .background(Color(0xFF121212))
                                .border(BorderStroke(1.dp, Color(0xFF1E1E1E)), RoundedCornerShape(12.dp))
                                .clickable { onNavigateToJoint(jointId) }
                                .padding(16.dp),
                        ) {
                            Row(verticalAlignment = Alignment.CenterVertically) {
                                Icon(Icons.Default.Hub, null, Modifier.size(14.dp), tint = Color(0xFF1E88E5))
                                Spacer(Modifier.width(6.dp))
                                Text(
                                    "ARTICULACIÓN RELACIONADA",
                                    style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif),
                                    fontWeight = FontWeight.ExtraBold,
                                    letterSpacing = (0.1f).sp,
                                    color = Color.White.copy(alpha = 0.5f),
                                )
                            }
                            Spacer(Modifier.height(8.dp))
                            Text(joint.name, style = MaterialTheme.typography.titleMedium.copy(fontFamily = FontFamily.Serif), fontWeight = FontWeight.Bold, color = Color(0xFF1E88E5))
                            Text(
                                WikiLabRepository.getJointTypeLabel(joint.type),
                                style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif),
                                color = Color.White.copy(alpha = 0.7f),
                            )
                        }
                    }
                }
            }

            // ─── Injuries ────────────────────────────────────────────────
            if (injuries.isNotEmpty()) {
                item {
                    TendonInjuriesCard(injuries, onNavigateToExercise)
                }
            }

            // ─── Protective Exercises ────────────────────────────────────
            if (protectiveExercises.isNotEmpty()) {
                item {
                    Column(
                        modifier = Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(12.dp))
                            .background(Color(0xFF121212))
                            .border(BorderStroke(1.dp, Color(0xFF1E1E1E)), RoundedCornerShape(12.dp))
                            .padding(16.dp),
                    ) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Icon(Icons.Default.Shield, null, Modifier.size(14.dp), tint = Color(0xFF43A047))
                            Spacer(Modifier.width(6.dp))
                            Text(
                                "EJERCICIOS PROTECTORES",
                                style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif),
                                fontWeight = FontWeight.ExtraBold,
                                letterSpacing = (0.1f).sp,
                                color = Color.White.copy(alpha = 0.5f),
                            )
                        }
                        Spacer(Modifier.height(8.dp))
                        protectiveExercises.forEach { exercise ->
                            Row(
                                modifier = Modifier.fillMaxWidth().clickable { onNavigateToExercise(exercise.id) }.padding(vertical = 5.dp),
                                verticalAlignment = Alignment.CenterVertically,
                            ) {
                                Surface(Modifier.size(8.dp), RoundedCornerShape(50), Color(0xFF43A047)) {}
                                Spacer(Modifier.width(10.dp))
                                Text(
                                    exercise.name,
                                    style = MaterialTheme.typography.bodyMedium.copy(fontFamily = FontFamily.Serif, color = Color.White),
                                    fontWeight = FontWeight.SemiBold,
                                    modifier = Modifier.weight(1f),
                                )
                                Icon(Icons.Default.ChevronRight, null, Modifier.size(16.dp), tint = Color(0xFF43A047).copy(alpha = 0.4f))
                            }
                        }
                    }
                }
            }

            // ─── TTC Recovery Info ───────────────────────────────────────
            item {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(12.dp))
                        .background(Color(0xFF121212))
                        .border(BorderStroke(1.dp, Color(0xFF1E1E1E)), RoundedCornerShape(12.dp))
                        .padding(16.dp),
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(Icons.Default.Timer, null, Modifier.size(16.dp), tint = Color(0xFFFF8F00))
                        Spacer(Modifier.width(6.dp))
                        Text(
                            "TIEMPO DE RECUPERACIÓN TENDINOSA",
                            style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif),
                            fontWeight = FontWeight.ExtraBold,
                            letterSpacing = (0.1f).sp,
                            color = Color.White.copy(alpha = 0.5f),
                        )
                    }
                    Spacer(Modifier.height(8.dp))
                    Text(
                        "Los tendones requieren de 6 a 12 semanas para adaptarse a cargas nuevas. La recuperación completa de una tendinopatía puede tomar de 3 a 6 meses.",
                        style = MaterialTheme.typography.bodySmall.copy(fontFamily = FontFamily.Serif),
                        color = Color.White.copy(alpha = 0.7f),
                        lineHeight = 18.sp,
                    )
                }
            }

            item { Spacer(Modifier.height(80.dp)) }
        }
    }
}

@Composable
private fun TendonInjuriesCard(
    injuries: List<InjuryInfo>,
    onNavigateToExercise: (String) -> Unit,
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(Color(0xFF121212))
            .border(BorderStroke(1.dp, Color(0xFF1E1E1E)), RoundedCornerShape(12.dp))
            .padding(16.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Icon(Icons.Default.Warning, null, Modifier.size(14.dp), tint = Color(0xFFE53935))
            Spacer(Modifier.width(6.dp))
            Text(
                "LESIONES COMUNES",
                style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif),
                fontWeight = FontWeight.ExtraBold,
                letterSpacing = (0.1f).sp,
                color = Color.White.copy(alpha = 0.5f),
            )
        }
        Spacer(Modifier.height(8.dp))

        injuries.forEachIndexed { i, injury ->
            Column(Modifier.fillMaxWidth().padding(vertical = 4.dp)) {
                Text(injury.name, style = MaterialTheme.typography.labelMedium.copy(fontFamily = FontFamily.Serif), fontWeight = FontWeight.Bold, color = Color(0xFFE53935))
                injury.description?.let { Text(it, style = MaterialTheme.typography.bodySmall.copy(fontFamily = FontFamily.Serif), color = Color.White.copy(alpha = 0.7f), lineHeight = 18.sp) }

                injury.contraindications?.let { contra ->
                    if (contra.isNotEmpty()) {
                        Spacer(Modifier.height(4.dp))
                        Text("Contraindicaciones:", style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif), fontWeight = FontWeight.Bold, color = Color(0xFFE53935))
                        contra.forEach { c -> Text("• $c", style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif), color = Color.White.copy(alpha = 0.7f)) }
                    }
                }

                injury.returnProgressions?.let { progs ->
                    if (progs.isNotEmpty()) {
                        Spacer(Modifier.height(4.dp))
                        Text("Progresión de retorno:", style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif), fontWeight = FontWeight.Bold, color = Color(0xFF43A047))
                        progs.forEachIndexed { j, p -> Text("${j + 1}. $p", style = MaterialTheme.typography.labelSmall.copy(fontFamily = FontFamily.Serif), color = Color(0xFF43A047)) }
                    }
                }
            }
            if (i < injuries.size - 1) HorizontalDivider(Modifier.padding(vertical = 6.dp), color = Color(0xFF1E1E1E))
        }
    }
}

File: ios-native/KPKNFit/KPKNFit/Presentation/Screens/Wikilab/TendonDetailScreen.swift
import SwiftUI

struct TendonDetailScreen: View {
    let tendonId: String
    let onNavigateToMuscle: (String) -> Void
    let onNavigateToJoint: (String) -> Void
    let onNavigateToExercise: (String) -> Void
    let onBack: () -> Void
    
    private var tendon: TendonEntity? { WikiLabRepository.shared.getTendonById(id: tendonId) }
    
    private var injuries: [InjuryInfo] {
        WikiLabRepository.shared.parseInjuries(jsonStr: tendon?.commonInjuries)
    }
    
    private var protectionIds: [String] {
        WikiLabRepository.shared.parseStringList(jsonStr: tendon?.protectiveExercises)
    }
    
    private var protectiveExercises: [WikiLabExerciseLink] {
        resolveWikiLabExerciseLinks(ids: protectionIds)
    }
    
    private var tendonGuide: WikiLabVisualGuide? {
        guard let t = tendon else { return nil }
        return buildTendonGuide(t)
    }
    
    var body: some View {
        ZStack {
            Color.black.ignoresSafeArea()
            
            if let tendon = tendon {
                List {
                    headerSection(tendon: tendon)
                    infoboxSection(tendon: tendon)
                    
                    if let guide = tendonGuide {
                        insightCardSection(guide: guide)
                    }
                    
                    relatedMuscleSection
                    relatedJointSection
                    injuriesSection
                    protectiveExercisesSection
                    recoverySection
                    
                    Color.clear.frame(height: 80)
                }
                .listStyle(.plain)
                .background(Color.black)
                .scrollContentBackground(.hidden)
            } else {
                Text("Tendón no encontrado")
                    .foregroundColor(.white)
            }
        }
        .navigationBarHidden(true)
    }
    
    private func headerSection(tendon: TendonEntity) -> some View {
        VStack(alignment: .leading, spacing: 8) {
            HStack(spacing: 8) {
                Button(action: onBack) {
                    Image(systemName: "chevron.left")
                        .font(.title3)
                        .foregroundColor(.white)
                }
                Text(tendon.name)
                    .font(.system(size: 32, weight: .black, design: .serif))
                    .foregroundColor(Color(hex: 0xFFFF8F00))
                Spacer()
            }
            
            if let desc = tendon.description {
                Text(desc)
                    .font(.system(size: 14, design: .serif))
                    .foregroundColor(.white.opacity(0.9))
                    .lineSpacing(4)
            }
        }
        .padding(.vertical, 8)
        .listRowInsets(EdgeInsets())
        .listRowBackground(Color.black)
    }
    
    private func infoboxSection(tendon: TendonEntity) -> some View {
        VStack(spacing: 8) {
            Text("Ficha Técnica Tendinosa")
                .font(.system(.title3, design: .serif))
                .fontWeight(.bold)
                .foregroundColor(.white)
                .frame(maxWidth: .infinity)
            Divider().background(Color(hex: 0x2C2C2C))
            
            InfoboxRowView(label: "Estructura", value: tendon.name)
            if let muscleId = tendon.muscleId {
                InfoboxRowView(label: "Músculo Asociado",
                              value: muscleId.replacingOccurrences(of: "_", with: " ").capitalized)
            }
            if let jointId = tendon.jointId {
                InfoboxRowView(label: "Articulación Asociada",
                              value: jointId.replacingOccurrences(of: "_", with: " ").capitalized)
            }
        }
        .padding(12)
        .background(Color(hex: 0x141414))
        .overlay(RoundedRectangle(cornerRadius: 4).stroke(Color(hex: 0x2C2C2C), lineWidth: 1))
        .listRowInsets(EdgeInsets())
        .listRowBackground(Color.black)
    }
    
    private func insightCardSection(guide: WikiLabVisualGuide) -> some View {
        WikiLabInsightCard(
            title: guide.title,
            accent: guide.accent,
            icon: guide.icon,
            summary: guide.summary,
            bullets: guide.bullets
        )
        .listRowInsets(EdgeInsets())
        .listRowBackground(Color.black)
    }
    
    private var relatedMuscleSection: some View {
        guard let muscleId = tendon?.muscleId else { return AnyView(EmptyView()) }
        let canonicalId = canonicalWikiLabMuscleIdFromEntityId(muscleId) ?? muscleId
        guard let muscle = WikiLabRepository.shared.getMuscleById(id: canonicalId) else { return AnyView(EmptyView()) }
        
        return AnyView(
            VStack(alignment: .leading, spacing: 8) {
                WikiSectionHeaderView(title: "Músculo Relacionado")
                HStack(spacing: 10) {
                    Circle()
                        .fill(Color(hex: 0x9C27B0))
                        .frame(width: 6, height: 6)
                    VStack(alignment: .leading, spacing: 2) {
                        Text(muscle.name)
                            .font(.system(size: 14, design: .serif))
                            .foregroundColor(Color(hex: 0x9C27B0))
                            .fontWeight(.bold)
                        Text(String(muscle.description.prefix(100)) + "...")
                            .font(.system(size: 13, design: .serif))
                            .foregroundColor(.white.opacity(0.6))
                    }
                }
                .padding(.leading, 12)
                .contentShape(Rectangle())
                .onTapGesture { onNavigateToMuscle(canonicalId) }
            }
            .listRowInsets(EdgeInsets())
            .listRowBackground(Color.black)
        )
    }
    
    private var relatedJointSection: some View {
        guard let jointId = tendon?.jointId else { return AnyView(EmptyView()) }
        guard let joint = WikiLabRepository.shared.getJointById(id: jointId) else { return AnyView(EmptyView()) }
        
        return AnyView(
            VStack(alignment: .leading, spacing: 8) {
                WikiSectionHeaderView(title: "Articulación Relacionada")
                HStack(spacing: 10) {
                    Circle()
                        .fill(Color(hex: 0x1E88E5))
                        .frame(width: 6, height: 6)
                    VStack(alignment: .leading, spacing: 2) {
                        Text(joint.name)
                            .font(.system(size: 14, design: .serif))
                            .foregroundColor(Color(hex: 0x1E88E5))
                            .fontWeight(.bold)
                        Text(WikiLabRepository.shared.getJointTypeLabel(type: joint.type))
                            .font(.system(size: 13, design: .serif))
                            .foregroundColor(.white.opacity(0.6))
                    }
                }
                .padding(.leading, 12)
                .contentShape(Rectangle())
                .onTapGesture { onNavigateToJoint(jointId) }
            }
            .listRowInsets(EdgeInsets())
            .listRowBackground(Color.black)
        )
    }
    
    private var injuriesSection: some View {
        guard !injuries.isEmpty else { return AnyView(EmptyView()) }
        
        return AnyView(
            VStack(alignment: .leading, spacing: 8) {
                WikiSectionHeaderView(title: "Lesiones Comunes y Patología")
                ForEach(injuries, id: \.id) { injury in
                    VStack(alignment: .leading, spacing: 4) {
                        Text(injury.name)
                            .font(.system(.subheadline, design: .serif))
                            .fontWeight(.bold)
                            .foregroundColor(Color(hex: 0xE53935))
                        
                        if let desc = injury.description {
                            Text(desc)
                                .font(.system(size: 14, design: .serif))
                                .foregroundColor(.white.opacity(0.8))
                                .lineSpacing(4)
                        }
                        
                        if let contra = injury.contraindications, !contra.isEmpty {
                            Text("Contraindicaciones:")
                                .font(.system(size: 11, design: .serif))
                                .fontWeight(.bold)
                                .foregroundColor(Color(hex: 0xE53935))
                            ForEach(contra, id: \.self) { c in
                                Text("• \(c)")
                                    .font(.system(size: 13, design: .serif))
                                    .foregroundColor(.white.opacity(0.7))
                            }
                        }
                        
                        if let progs = injury.returnProgressions, !progs.isEmpty {
                            Text("Progresión de retorno:")
                                .font(.system(size: 11, design: .serif))
                                .fontWeight(.bold)
                                .foregroundColor(Color(hex: 0x43A047))
                            ForEach(Array(progs.enumerated()), id: \.offset) { j, p in
                                Text("\(j + 1). \(p)")
                                    .font(.system(size: 13, design: .serif))
                                    .foregroundColor(Color(hex: 0x43A047))
                            }
                        }
                    }
                    .padding(.leading, 12)
                    .padding(.vertical, 4)
                }
            }
            .listRowInsets(EdgeInsets())
            .listRowBackground(Color.black)
        )
    }
    
    private var protectiveExercisesSection: some View {
        guard !protectiveExercises.isEmpty else { return AnyView(EmptyView()) }
        
        return AnyView(
            VStack(alignment: .leading, spacing: 8) {
                WikiSectionHeaderView(title: "Ejercicios Protectores")
                ForEach(protectiveExercises, id: \.id) { exercise in
                    HStack(spacing: 10) {
                        Circle()
                            .fill(Color(hex: 0x66BB6A))
                            .frame(width: 6, height: 6)
                        Text(exercise.name)
                            .font(.system(size: 14, design: .serif))
                            .foregroundColor(Color(hex: 0x66BB6A))
                            .fontWeight(.bold)
                        if !exercise.subtitle.isEmpty {
                            Text("· \(exercise.subtitle)")
                                .font(.system(size: 13, design: .serif))
                                .foregroundColor(.white.opacity(0.5))
                        }
                    }
                    .padding(.leading, 12)
                    .padding(.vertical, 2)
                    .contentShape(Rectangle())
                    .onTapGesture { onNavigateToExercise(exercise.id) }
                }
            }
            .listRowInsets(EdgeInsets())
            .listRowBackground(Color.black)
        )
    }
    
    private var recoverySection: some View {
        VStack(alignment: .leading, spacing: 8) {
            WikiSectionHeaderView(title: "Tiempos de Adaptación y Recuperación")
            Text("Los tendones tienen una tasa metabólica más baja que los músculos y requieren de 6 a 12 semanas para adaptarse estructuralmente a cargas nuevas de entrenamiento. En caso de una tendinopatía o sobrecarga, la recuperación fisiológica completa suele tomar de 3 a 6 meses de fortalecimiento progresivo.")
                .font(.system(size: 14, design: .serif))
                .foregroundColor(.white.opacity(0.8))
                .lineSpacing(4)
                .padding(.leading, 12)
        }
        .listRowInsets(EdgeInsets())
        .listRowBackground(Color.black)
    }
}

private func WikiSectionHeaderView(title: String) -> some View {
    VStack(alignment: .leading, spacing: 4) {
        Text(title)
            .font(.system(.title3, design: .serif))
            .fontWeight(.black)
            .foregroundColor(.white)
        Divider().background(Color(hex: 0x2C2C2C))
    }
    .padding(.top, 12)
}

private func InfoboxRowView(label: String, value: String) -> some View {
    HStack {
        Text(label)
            .font(.system(size: 14, design: .serif))
            .fontWeight(.bold)
            .foregroundColor(.white.opacity(0.5))
        Spacer()
        Text(value)
            .font(.system(size: 14, design: .serif))
            .fontWeight(.medium)
            .foregroundColor(.white)
    }
}
